### PR TITLE
Failing test for HTTP username

### PR DESCRIPTION
**Pull request by fregante**
Example (noticeable with Refined GitHub): https://github.com/scieloorg/opac_proc/blob/46d9f826d2519573b99418f11da1ce3f87bfa7eb/requirements.txt#L12-L14

Related: should it support `git+http` urls at all?

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ test('DocumentFragment support', t => {
 
 test('supports `@` in the URL path', t => {
 	t.is(m('https://sindresorhus.com/@foo'), '<a href="https://sindresorhus.com/@foo">https://sindresorhus.com/@foo</a>');
+	t.is(m('https://user@sindresorhus.com/@foo'), '<a href="https://user@sindresorhus.com/@foo">https://user@sindresorhus.com/@foo</a>');
```
Comment (kevva) [inline test.js]: `t.is` => `t.fail`

**Review by kevva — CHANGES_REQUESTED** [review]

File: test.js [hunk 2]
```diff
@@ -90,6 +90,7 @@ test('DocumentFragment support', t => {
 
 test('supports `@` in the URL path', t => {
 	t.is(m('https://sindresorhus.com/@foo'), '<a href="https://sindresorhus.com/@foo">https://sindresorhus.com/@foo</a>');
+	t.fail(m('https://user@sindresorhus.com/@foo'), '<a href="https://user@sindresorhus.com/@foo">https://user@sindresorhus.com/@foo</a>');
```
Comment (kevva) [inline test.js]: Sorry, my bad 😿 . This line needs to be in a separate `test`:

```js
test.failing('supports username in url', t => {
    t.is(m('https://user@sindresorhus.com/@foo'), '<a href="https://user@sindresorhus.com/@foo">https://user@sindresorhus.com/@foo</a>');
});
```

**Review by kevva — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
> Related: should it support git+http urls at all?

Is there really any point? Clicking it wouldn't work anyway.

**Comment by fregante** [comment]
@sindresorhus it's mostly about expectations. Also perhaps people have git URL handlers that open SourceTree for example.

**Comment by sindresorhus** [comment]
Who's' expectations though? I definitely wouldn't expect such URLs to be linkified. I actually don't want it at all, as it creates noise and gives a false expectation that they will work.

> Also perhaps people have git URL handlers that open SourceTree for example.

That would be a minority, and everyone else would have to pay with useless links and wasted time clicking them and then realizing it's not working.

**Comment by sindresorhus** [comment]
I see that `-e git+https://github.com/scieloorg/xylose@1.17.5#egg=xylose` got linkified in your referenced example. I don't think we should linkify that either as it obviously doesn't work. In your linked case, it would make more sense for Octolinker to handle those.

**Comment by fregante** [comment]
How do we detect that though? The regex will have to look for the character that comes before `http` as well

**Comment by sindresorhus** [comment]
We would need a lookbehind regex pattern, which is still just a draft standard. Alternatively we could change it to also match `git+` and then filter the URLs afterwards.

**Comment by sindresorhus** [comment]
@bfred-it Could you change the test to what we've discussed? (Not allowing + prefixed URLs)

**Comment by fregante** [comment]
Not sure of how to do that :D

**Comment by sindresorhus** [comment]
I'm only talking about the test, not the implementation. This is what I meant: https://github.com/sindresorhus/linkify-urls/commit/4db31ab79d62dc0adf6b3c086a6a68f955343918 :)

**Comment by sindresorhus** [comment]
Continued in https://github.com/sindresorhus/linkify-urls/issues/7

**Comment by fregante** [comment]
That’s unrelated to this PR, URLs with user:pass are still real HTTP and browsers can handle them

**Comment by sindresorhus** [comment]
Oops, I was too focused on the + stuff.

**Pull request merged**